### ISSUE TITLE
Workflows duplicates on pull request

### ISSUE THREAD

**Issue by foo2810** [foo2810/ad-hoc-vimconfig#17]
On pull request, both push and pull request event are triggered.
Duplicate workflow execution is wasteful.

**Comment by foo2810** [comment]
**Root cause**

During opening pull request at source branch of devel branch, when new commits are pushed, both push and pull_request events are triggered.
The `synchronize` event is main cause.

```yaml
on:
  push:
    branches: ["main", "devel"]
  pull_request:
    branches: [ "*" ]
    types: ["opened", "synchronize", "reopened"]
```

So, redefining new workflow of pull_request event for devel branches, the problem should be resolved.

**Comment by foo2810** [comment]
To define the new workflows, [Reusing workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows) may be utilized.

**Comment by foo2810** [comment]
If the synchronize event of a pull request was suppressed only when the pull request is opened in the devel branch, the workflow implementation would be complicated.

This is because the workflow changes depending on the version of vim.

If CI tests are executed only on push events to the main branch and on pull request events (i.e., not execute tests on push events to the devel branch), the problem can be avoided, and the workflow implementation will be less complicated.

**Comment by foo2810** [comment]
Implemented at #18 and merged into main.
